kusto/query_builder.go

```diff
@@ -506,6 +435,166 @@ func NewStmt(query stringConstant, options ...StmtOption) Stmt {
+// AddTable - Given a variable with a value 'MyTable', AddTable(arg) will produce the following string
+// '["MyTable"]' and will append the following string to Stmt
+func (s Stmt) AddTable(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddColumn - Given a variable with a value 'MyColumn', AddColumn(arg) will produce the following string
+// '["MyColumn"]' and will append the following string to Stmt
+func (s Stmt) AddColumn(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// AddFunction - Given a variable with a value 'MyFunction', AddFunction(arg) will produce the following string
+// '["MyFunction"]' and will append the following string to Stmt
+func (s Stmt) AddFunction(query string) Stmt {
+	return s.NormalizeName(query, false)
+}
+
+// NormalizeName normalizes a string in order to be used safely in the engine - given "query" will produce [\"query\"].
+func (s Stmt) NormalizeName(query string, forceNormalization bool) Stmt {
+	if query == "" {
+		return s
+	}
+	if !forceNormalization && !RequiresQuoting(query) {
+		s.queryStr = s.queryStr + query
+		return s
+	}
+
+	s.queryStr = s.queryStr + "[" + AddQuotedString(query, false) + "]"
+	return s
+}
+
+// RequiresQuoting checks whether a given string is an identifier
+func RequiresQuoting(query string) bool {
+	if query == "" {
+		return true
+	}
+	if !unicode.IsLetter(rune(query[0])) && rune(query[0]) != '_' {
+		return true
+	}
+	for _, c := range query {
+		if !(unicode.IsLetter(c) || unicode.IsDigit(c) || c == '_') {
+			return true
+		}
+	}
+	return false
+}
+
+// AddQuotedString escapes a string to be safely added to a stmt
+func AddQuotedString(value string, hidden bool) string {
+	if value == "" {
+		return value
+	}
+
+	var literal strings.Builder
+
+	if hidden {
+		literal.WriteString("h")
+	}
+	literal.WriteString("\"")
+	for _, c := range value {
+		switch c {
+		case '\'':
+			literal.WriteString("\\'")
+
+		case '"':
+			literal.WriteString("\\\"")
+
+		case '\\':
+			literal.WriteString("\\\\")
+
+		case '\x00':
+			literal.WriteString("\\0")
+
+		case '\a':
+			literal.WriteString("\\a")
+
+		case '\b':
+			literal.WriteString("\\b")
+
+		case '\f':
+			literal.WriteString("\\f")
+
+		case '\n':
+			literal.WriteString("\\n")
+
+		case '\r':
+			literal.WriteString("\\r")
+
+		case '\t':
+			literal.WriteString("\\t")
+
+		case '\v':
+			literal.WriteString("\\v")
+
+		default:
+			if !ShouldBeEscaped(c) {
+				literal.WriteString(string(c))
+			} else {
+				literal.WriteString(fmt.Sprintf("\\u%04x", c))
+			}
+
+		}
+	}
+	literal.WriteString("\"")
+
+	return literal.String()
+}
+
+// ShouldBeEscaped Checks whether a rune should be escaped or not based on it's type.
+func ShouldBeEscaped(c int32) bool {
+	if c <= unicode.MaxLatin1 {
+		return unicode.IsControl(c)
+	}
+	return true
+}
+
+// AddInt will add an int as a string to the Stmt.  This allows dynamically building of a query from a root Stmt.
+func (s Stmt) AddInt(query int) Stmt {
```

Comment:
AddInt32,and AddInt64 (for long)

Reply:
As our discussion above - i think int64 is only for long. and we can also add a addint32 - but i'll admit i dont think it will be used, most likely addint will be the "goto" one. WDYT?